python/lsst/ts/salobj/sal_info.py

```diff
@@ -396,6 +437,12 @@ def assert_started(self) -> None:
         if not self.started:
             raise RuntimeError("Not started")
 
+    def assert_running(self) -> None:
+        """Raise RuntimeError if not running."""
+        if not self.running:
+            msg = "Not started" if not self.started else "No longer open"
+            raise RuntimeError(msg)
```

Comment:
I agree. However, I strongly prefer to raise RuntimeError, rather than AssertionError, if a user tries to write using a non-running SalInfo. So I think the potential confusion is justified. I could call it `raise_runtime_error_if_not_running` but I'd prefer to keep it as is.

This is a common pattern in salobj. BaseCsc has assert_enabled which raises salobj.ExpectedError, in order to avoid a traceback when a command is rejected because the CSC is not enabled.